src/app/Core/calculateRowsWithPercent.service.ts

```diff
@@ -0,0 +1,43 @@
+function getSumOfIsolates(objList: Record<string, string>[]): number {
+    let sum = 0;
+    objList.forEach((obj) => {
+        const k = Object.keys(obj);
+        k.forEach((element) => {
+            if (element !== "name") {
+                sum += (obj[element] as unknown) as number;
+            }
+        });
+    });
+    return sum;
+}
+
+function calculatePercent(part: number, total: number): string {
+    const percent = (part / total) * 100;

```

Comment:
reject: handle the case if total = 0, to prevent this function from throwing an exception. Or if 0 is forbidden explicitly throw an exception if total = 0